internal/test/envoy.go

```diff
@@ -42,19 +42,26 @@ type Runner interface {
 // RequireRun executes Run on the given Runtime and calls shutdown after it started.
 func RequireRun(t *testing.T, shutdown func(), r Runner, stderr io.Reader, args ...string) (err error) {
 	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	if shutdown == nil {
 		shutdown = cancel
 	}
+
 	go func() {
 		err = r.Run(ctx, args)
 		cancel()
 	}()
 
 	reader := bufio.NewReader(stderr)
-	require.Eventually(t, func() bool {
-		b, err := reader.Peek(512)
-		return err != nil && strings.Contains(string(b), "initializing epoch 0")
-	}, 2*time.Second, 100*time.Millisecond, "never started process")
+	waitFor := "initializing epoch 0"
+	if !assert.Eventually(t, func() bool {
+		b, e := reader.Peek(512)
+		return e != nil && strings.Contains(string(b), waitFor)
+	}, 5*time.Second, 100*time.Millisecond) {
```

Comment:
I switched this after finding out some were legit timeouts and the process didn't finish, yet. Using an "if" allows us to format state before the error raises